Source/DafnyTestGeneration/ProgramModification.cs

```diff
@@ -131,25 +156,35 @@ private enum Status { Success, Failure, Untested }
         return counterexampleLog;
       }
       var log = writer.ToString();
+      // If Dafny finds several assertion violations (e.g. because of inlining one trap assertion multiple times),
+      // pick the first execution trace and extract the counterexample from it
```

Comment:
Is this always the right trace?

Reply:
The choice of the trace shouldn't matter since I replace all native assertions with assumptions and so the only assertion that can fail is the one that is being targeted. One can imagine an optimization that picks a trace covering the most blocks but in terms correctness the choice should not affect the result.